src/file.cc

```diff
@@ -222,65 +222,74 @@ void write(int fd, StringView data)
 
 void write_buffer_to_fd(Buffer& buffer, int fd)
 {
-    auto eolformat = buffer.options()["eolformat"].get<EolFormat>();
-    StringView eoldata;
-    if (eolformat == EolFormat::Crlf)
-        eoldata = "\r\n";
-    else
-        eoldata = "\n";
-
-    if (buffer.options()["BOM"].get<ByteOrderMark>() == ByteOrderMark::Utf8)
-        if (::write(fd, "\xEF\xBB\xBF", 3) < 0)
-            throw runtime_error(format("unable to write data to the buffer (fd: {}; errno: {})", fd, ::strerror(errno)));
-
-    for (LineCount i = 0; i < buffer.line_count(); ++i)
+    if ((buffer.flags() & Buffer::Flags::ReadOnly) == false)
```

Comment:
I prefer early returns as they avoid a needless indentation, and I dont see a big problem with `return;` in void functions, we are not returning a value, but we are returning (early) from the function.

Thats a matter of taste, however at least for consistency could you convert that to just an early return ?